src/test/unit/EigenPodManagerUnit.t.sol

```diff
@@ -117,220 +85,438 @@ contract EigenPodManagerUnitTests is Test, EigenPodPausingConstants {
-    //         queuedWithdrawal.delegatedAddress,
-    //         queuedWithdrawal.withdrawer,
-    //         withdrawalRoot
-    //     );
-    //     eigenPodManager.completeQueuedWithdrawal(queuedWithdrawal, middlewareTimesIndex);
-    //     cheats.stopPrank();
-
-    //     // TODO: make EigenPodMock do something so we can verify that it gets called appropriately?
-    //     uint256 eigenPodBalanceAfter = address(eigenPod).balance;
-
-    //     // verify that the withdrawal root does bit exist after queuing
-    //     require(!eigenPodManager.withdrawalRootPending(withdrawalRoot), "withdrawalRootPendingBefore is true!");
-    // }
-
-    // INTERNAL / HELPER FUNCTIONS
-    // deploy an EigenPod for the staker and check the emitted event
-    function _deployEigenPodForStaker(address staker) internal returns (IEigenPod deployedPod) {
+    function _deployAndReturnEigenPodForStaker(address staker) internal returns (IEigenPod deployedPod) {
         deployedPod = eigenPodManager.getPod(staker);
-        cheats.startPrank(staker);
-        cheats.expectEmit(true, true, true, true, address(eigenPodManager));
-        emit PodDeployed(address(deployedPod), staker);
+        cheats.prank(staker);
         eigenPodManager.createPod();
-        cheats.stopPrank();
         return deployedPod;
     }
 
-// TODO: reimplement similar test
-    // // creates a queued withdrawal of "beacon chain ETH shares", from `staker`, of `amountWei`, "to" the `withdrawer`
-    // function _createQueuedWithdrawal(address staker, uint256 amountWei, address withdrawer)
-    //     internal
-    //     returns (IEigenPodManager.BeaconChainQueuedWithdrawal memory queuedWithdrawal, bytes32 withdrawalRoot)
-    // {
-    //     // create the struct, for reference / to return
-    //     queuedWithdrawal = IEigenPodManager.BeaconChainQueuedWithdrawal({
-    //         shares: amountWei,
-    //         podOwner: staker,
-    //         nonce: eigenPodManager.cumulativeWithdrawalsQueued(staker),
-    //         startBlock: uint32(block.number),
-    //         delegatedTo: delegationManagerMock.delegatedTo(staker),
-    //         withdrawer: withdrawer
-    //     });
-
-    //     // verify that the withdrawal root does not exist before queuing
-    //     require(!eigenPodManager.withdrawalRootPending(withdrawalRoot), "withdrawalRootPendingBefore is true!");
-
-    //     // get staker nonce and shares before queuing
-    //     uint256 nonceBefore = eigenPodManager.cumulativeWithdrawalsQueued(staker);
-    //     int256 sharesBefore = eigenPodManager.podOwnerShares(staker);
-
-    //     // actually create the queued withdrawal, and check for event emission
-    //     cheats.startPrank(staker);
+    function _checkPodDeployed(address staker, uint256 numPodsBefore) internal {
+        // Get expected pod address
+        IEigenPod expectedPod = eigenPodManager.getPod(staker);
+
+        assertEq(address(eigenPodManager.ownerToPod(staker)), address(expectedPod));
```

Comment:
seems vacuous -- since `eigenPodManager.getPod` reads from `eigenPodManager.ownerToPod`, this will always pass. seems like we'd need to store the `expectedPod` _prior_ to deploying an EigenPod, if we wanted to check what I think this is designed to check

Reply:
Yup, good catch - updated order